### ISSUE TITLE
TextEdit always clips content

### ISSUE THREAD

**Issue by somnivore** [godotengine/godot#32996]
**Godot version:**
3564547208030264a28cc55232992aab3cbdad94

**OS/device including version:**
ArcoLinux

**Issue description:**
TextEdit forcibly clips all content, regardless of whether the `Clip Content` box is checked.

![image](https://user-images.githubusercontent.com/28724859/67325878-6faaf200-f4db-11e9-8bde-2ea76b8f37fa.png)
![image](https://user-images.githubusercontent.com/28724859/67325902-78032d00-f4db-11e9-8e1a-2bc4a4a0d35e.png)

**Comment by YeldhamDev** [comment]
I'm quite sure `TextEdit` wouldn't even work properly without clipping.

**Comment by codecustard** [comment]
Clip Content should probably default to true in `TextEdit` and the following line should be changed to reflect the clip content property:
https://github.com/godotengine/godot/blob/72dfa67dd3508f20592959d306e8f2ef159eea5f/scene/gui/text_edit.cpp#L725

**Comment by girng** [comment]
Should probably use a different node or at least put that TextureRect inside another node and container it

**Comment by mitsuan** [comment]
can I contribute to this project please.

**Comment by ondesic** [comment]
I agree that it should be set to TRUE be default, but allow the user to turn off clipping.  As a side not, LineEdit allows no clipping, and it looks great.

**Comment by Calinou** [comment]
I can confirm this on commit 4.0.dev.custom_build [b5b633ff5]. In particular, I need this to display a focus outline that's slightly *outside* the TextEdit. Removing the clipping function call in the TextEdit C++ code works, but it likely has unintended side-effects :slightly_smiling_face:

**Comment by KoBeWi** [comment]
This happens when you disable clipping:
![image](https://user-images.githubusercontent.com/2223172/146410412-36a08b33-4565-405b-a5d7-a89d0514b76c.png)
Clipping is global for the CanvasItem, you can't disable it only for some parts. So to make this work properly, we'd need to move all text rendering to a separate canvas item (server-side). Similar thing is done for quadrants in TileMap.

**Comment by kitbdev** [comment]
- Clip contents can be disabled as of https://github.com/godotengine/godot/pull/89993
